File: simple/csharp/introduce-null-object_before2.cs
void initConnection() 
{
  if (channel == null) 
  {
    throw new System.ArgumentException("Communication channel must be specified");
  }

  switch (customer) 
  {
    case INTERNET:
      connection = new HihgSpeedConnection();
      break;
    case HOME_PHONE:
      connection = new HomeConnection();
      break;
    case CELL_PHONE:
      connection = new LTEConnection();
  }
}

File: simple/php/introduce-null-object_before2.php
function initConnection() {
  if (channel == null) {
    throw new InvalidArgumentException('Communication channel must be specified');
  }

  switch ($customer) {
    case $INTERNET:
      $connection = new HihgSpeedConnection();
      break;
    case $HOME_PHONE:
      $connection = new HomeConnection();
      break;
    case $CELL_PHONE:
      $connection = new LTEConnection();
  }
}
